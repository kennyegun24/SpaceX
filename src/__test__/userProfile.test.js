import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/configureStore';

import Profile from '../pages/userProfile';

describe('Tests for the missions component/pag', () => {
  const renderElement = () => render(
    <Provider store={store}>
      <Router>
        <Profile />
      </Router>
    </Provider>,
  );

  it('Should render The userProfile page properly', () => {
    const profile = renderElement();
    expect(profile).toMatchSnapshot();
  });
});
