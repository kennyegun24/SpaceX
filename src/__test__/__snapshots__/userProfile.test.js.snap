// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the missions component/pag Should render The userProfile page properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="user-container"
      >
        <div
          class="missions-column"
        >
          <header>
            My Missions
          </header>
          <div
            class="no__reservations"
          >
            <p>
              No Reservations Made
            </p>
            <a
              class="reserve-link"
              href="/missions"
            >
              Click to make Reservations
            </a>
          </div>
        </div>
        <div
          class="rockets-column"
        >
          <header>
            My Rockets
          </header>
          <div
            class="no__reservations"
          >
            <p>
              No Reservations Made
            </p>
            <a
              class="reserve-link"
              href="/"
            >
              Click to make Reservations
            </a>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="user-container"
    >
      <div
        class="missions-column"
      >
        <header>
          My Missions
        </header>
        <div
          class="no__reservations"
        >
          <p>
            No Reservations Made
          </p>
          <a
            class="reserve-link"
            href="/missions"
          >
            Click to make Reservations
          </a>
        </div>
      </div>
      <div
        class="rockets-column"
      >
        <header>
          My Rockets
        </header>
        <div
          class="no__reservations"
        >
          <p>
            No Reservations Made
          </p>
          <a
            class="reserve-link"
            href="/"
          >
            Click to make Reservations
          </a>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
